h5p_proplist.go

```diff
@@ -58,8 +69,15 @@ func (p *PropList) SetChunk(dim []uint) error {
 }
 
 // SetDeflate sets deflate (GNU gzip) compression method and compression level.
+// If level is set as DefaultCompression, 6 will be used.
 // https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-SetDeflate
-func (p *PropList) SetDeflate(level uint) error {
+func (p *PropList) SetDeflate(level int) error {
+	if level == DefaultCompression {
+		level = 6
+	}
+	if level < 0 {
```

Comment:
Check for grater than 9 as well.

Reply:
Or BestCompression.

Reply:
It will be checked in C.

Reply:
If that's the case, there is no need to check whether `level < 0`.

Reply:
The `level` in C is `uint`. A minus value will be meaningless after converted to `uint`. In this case, Go API user cannot understand the returning error from C.

Reply:
This is not true. You are performing a `C.uint(level)` conversion. This will give you `^C.uint(0)`. If we pretend for a moment that `uint` and `C.uint` are the same, then we can see that [here](https://play.golang.org/p/TTmG7d7AvUy).

If `0xffffffff` is meaningless, then so is 9.

Reply:
"Meaningless" I said is not proper here. It should be "clear". If I set level as 12, error will be "unsupported level 12". But if I set level as -3, it will be "unsupported level 4294967293". It not clear for the user.

Reply:
There is no error indicating the actual invalid value if it is discovered by HDF5. All we get back is an error code, -1, on failure, according to the HDF5 docs.

Please add the check for `> BestCompression`. When you have done that and sent a PR to gonum/gonum adding yourself to A+C with the commit message "A+C: add <yourfullname>" I will approve this PR. I don't want to waste more time here.

Reply:
Also the unsafe code above.